lib/tpm2_auth_util.c

```diff
@@ -48,8 +50,91 @@ static bool handle_hex_password(const char *password, TPM2B_AUTH *auth) {
     return true;
 }
 
+static bool handle_password(const char *password, TPM2B_AUTH *auth);
+
+static tool_rc get_auth_for_file_param(const char* password, TPM2B_AUTH *auth) {
+    const char* path = password;
+    size_t size = (sizeof(auth->buffer) * 2) + HEX_PREFIX_LEN + 1;
+    bool is_a_tty = isatty(STDIN_FILENO);
+    UINT8 *buffer = calloc(1, size);
+
+    if (!buffer) {
+        LOG_ERR("oom");
+        return tool_rc_general_error;
+    }
+
+    if (path) {
+        path += FILE_PREFIX_LEN;
+        path = strcmp("-", path) ? path : NULL;
+    }
+    if (!is_a_tty || path) {
+        UINT16 fsize = sizeof(buffer) - 1;
```

Comment:
`buffer` is not an array but a heap pointer.  `fsize` will be `7` on 64-bit systems and `3` on 32-bit systems.